[Mailer] Fix `Trying to access array offset on value of type null` error by adding null checking

khushaalan opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix `Trying to access array offset on value of type null` error by adding null checking
| License       | MIT

Its a very simple bug fix.. basically the `error_get_last()` doesn't have the 'message' and causing this issue:
![image](https://github.com/user-attachments/assets/95b51b87-0011-4913-a67a-ce91195d4f2e)

so i just added Null Checking and it seems to be working now:
![image](https://github.com/user-attachments/assets/0d2f7dae-8854-421c-a946-10426b15ee06)

this is happens when using Mailtrap and when it reaches its quota..

nicolas-grekas commented on src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php:
```diff
@@ -88,7 +88,12 @@ public function readLine(): string
+                if ($this->getReadConnectionDescription() && error_get_last()) {
+                    $error = $this->getReadConnectionDescription().error_get_last()['message'];
+                } else {
+                    $error = $this->getReadConnectionDescription().error_get_last();
```
this line doesn't make sense sorry, error_get_last return an array for example.
no need to check getReadConnectionDescription IMHO, it has to return something

I'm wondering also: would calling error_get_last before calling getReadConnectionDescription could lead to a non empty message?

khushaalan replied:
should i just check for isset(error_get_last()) instead? The goal is to just check the error_get_last() have 'message' object or not. Without this checking, it doesnt throw the TransportException. instead just throws the array offset error.

khushaalan replied:
@nicolas-grekas any updates on this

nicolas-grekas reviewed (commented):
(Bugfix for 6.4 when merging)

nicolas-grekas committed:
bug #60094 [DoctrineBridge] Fix support for entities that leverage native lazy objects (nicolas-grekas)

This PR was merged into the 6.4 branch.

Discussion
----------

[DoctrineBridge] Fix support for entities that leverage native lazy objects

| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Commits
-------

382b3dd333d [DoctrineBridge] Fix support for entities that leverage native lazy objects

xabbuh commented on src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php:
```diff
@@ -88,7 +88,12 @@ public function readLine(): string
                 throw new TransportException(\sprintf('Connection to "%s" has been closed unexpectedly.', $this->getReadConnectionDescription()));
             }
             if (false === $line) {
-                throw new TransportException(\sprintf('Unable to read from connection to "%s": ', $this->getReadConnectionDescription()).error_get_last()['message']);
```
```php
throw new TransportException(\sprintf('Unable to read from connection to "%s": ', $this->getReadConnectionDescription()).(error_get_last()['message'] ?? ''));
```

What about changing this line to this instead?

khushaalan replied:
that works as well. ok i update already

fabpot commented:
Thank you @khushaalan.
